core/build_playerglobal/src/lib.rs

```diff
@@ -479,9 +475,39 @@ fn write_native_table(data: &[u8], out_dir: &Path) -> Result<Vec<u8>, Box<dyn st
                     }
                 }
             }
-            TraitKind::Method { method, .. }
-            | TraitKind::Getter { method, .. }
-            | TraitKind::Setter { method, .. } => {
+            TraitKind::Method { disp_id, method }
+            | TraitKind::Getter { disp_id, method }
+            | TraitKind::Setter { disp_id, method } => {
+                if trait_has_metadata(&abc, trait_, METADATA_NATIVE_CALLABLE) {
+                    if disp_id == 0 {
+                        panic!("ASC should calculate disp ids for all methods; cannot apply NativeCallable without a compiler-calculated disp id")
+                    } else {
+                        // Disp-ids are 1-indexed, but ASC generates them two disp-ids
+                        // off. Instead of subtracting 1 from it the disp-id, add 1.
+                        let disp_id = disp_id + 1;
+
+                        let (trait_name, mut const_name) =
+                            rust_path_and_trait_name(&abc, trait_, parent);
+
+                        // Make getters prefixed with GET_ and setters prefixed with SET_
+                        if matches!(trait_.kind, TraitKind::Getter { .. }) {
+                            const_name = "GET_".to_owned() + &const_name;
+                        } else if matches!(trait_.kind, TraitKind::Setter { .. }) {
+                            const_name = "SET_".to_owned() + &const_name;
+                        }
+
+                        let const_name = quote::format_ident!("{}", const_name);
```

Comment:
This could be written more succintly as:
```rs
let const_name = match trait_.kind {
    TraitKind::Getter => format_ident!("GET_{}", const_name),
    TraitKind::Setter => format_ident!("SET_{}", const_name),
    _ => format_ident!("{}", const_name),
};